vms/proposervm/height_indexed_vm.go

```diff
@@ -85,31 +33,6 @@ func (vm *VM) GetBlockIDAtHeight(ctx context.Context, height uint64) (ids.ID, er
 	}
 }
 
-// As postFork blocks/options are accepted, height index is updated even if its
-// repairing is ongoing. vm.ctx.Lock should be held
-func (vm *VM) updateHeightIndex(height uint64, blkID ids.ID) error {
-	_, err := vm.State.GetCheckpoint()
-	switch err {
-	case nil:
-		// Index rebuilding is ongoing. We can update the index with the current
-		// block.
-
-	case database.ErrNotFound:
-		// No checkpoint means indexing has either not started or is already
-		// done.
-		if !vm.hIndexer.IsRepaired() {
-			return nil
-		}
-
-		// Indexing must have finished. We can update the index with the current
-		// block.
-
-	default:
-		return fmt.Errorf("failed to load index checkpoint: %w", err)
-	}
-	return vm.storeHeightEntry(height, blkID)
-}
```

Comment:
The only part of this function remaining is `storeHeightEntry` so this function is deleted in favor of calling that directly.

Can rename that to `updateHeightIndex` if preferred.